Scalafmt 3.5.3: Unexpected do-while formatting

- https://github.com/wartremover/wartremover/pull/637

## Configuration (required) ##

Please paste the contents of your `.scalafmt.conf` file here:

```
version = "3.5.4"
runner.dialect = Scala213
maxColumn = 120
align.preset = none
align.tokens = []
rewrite.rules = [
  RedundantParens
  ExpandImportSelectors
  PreferCurlyFors
]
rewrite.imports.contiguousGroups = "no"
rewrite.imports.groups = [[".*"]]
continuationIndent.callSite = 2
continuationIndent.defnSite = 2
docstrings.style = keep
includeCurlyBraceInSelectChains = false
optIn.breakChainOnFirstMethodDot = false
trailingCommas = preserve
newlines.topLevelStatementBlankLines = [
  {
    blanks { after = 1 }
    maxNest = 1
    regex = "Import"
  }
]
project.layout = StandardConvention
```

## Command-line parameters ##

`sbt scalafmt`

## Steps

Given code like this:
```scala
      do {
        println()
      } while (true)
```

## Problem

Scalafmt formats code like this:
```scala
      do {
        println()
      } while true
```

```
[error] org.scalafmt.sbt.ScalafmtSbtReporter$ScalafmtSbtError: scalafmt: /home/runner/work/wartremover/wartremover/core/src/test/scala/org/wartremover/test/WhileTest.scala:23: error: [dialect scala213] ( expected but true found
[error]       } while true
[error]               ^ [/home/runner/work/wartremover/wartremover/core/src/test/scala/org/wartremover/test/WhileTest.scala]
```

## Expectation

don't remove parentheses

## Workaround

remove `rewrite.rules = [RedundantParens]`

## Notes

--- a/scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/RedundantParens.scala
+++ b/scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/RedundantParens.scala
@@ -97,6 +97,7 @@ class RedundantParens(ftoks: FormatTokens) extends FormatTokensRewrite.Rule {
           case _: Enumerator.Guard => RewriteCtx.isPostfixExpr(t)
           case p: Case =>
             p.cond.contains(t) && RewriteCtx.isPostfixExpr(t)
+          case _: Term.Do => false
           case p: Term.While =>
             style.dialect.allowSignificantIndentation && p.expr == t &&
             ftoks.tokenBefore(p.body).left.is[Token.KwDo]

--- a/scalafmt-tests/src/test/resources/rewrite/RedundantParens.stat
+++ b/scalafmt-tests/src/test/resources/rewrite/RedundantParens.stat
@@ -1025,3 +1025,12 @@ object a {
   // all
   foo || bar && baz // low precedence infix
 }
+<<< #3222 do-while
+object a {
+  do foo while (true)
+}
+>>>
+test does not parse
+object a {
+  do foo while true
+               ^

--- a/scalafmt-tests/src/test/resources/rewrite/RedundantParens.stat
+++ b/scalafmt-tests/src/test/resources/rewrite/RedundantParens.stat
@@ -1030,7 +1030,6 @@ object a {
   do foo while (true)
 }
 >>>
-test does not parse
 object a {
-  do foo while true
-               ^
+  do foo while (true)
+}

